ports/rp2/machine_uart.c

```diff
@@ -192,14 +305,28 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
         if (self->invert & UART_INVERT_TX) {
             gpio_set_outover(self->tx, GPIO_OVERRIDE_INVERT);
         }
+        // allocate the buffers
+        ringbuf_alloc(&(self->read_buffer), rxbuf_len + 1);
+        MP_STATE_PORT(rp2_uart_rx_buffer[uart_id]) = self->read_buffer.buf;
+
+        ringbuf_alloc(&(self->write_buffer), txbuf_len + 1);
+        MP_STATE_PORT(rp2_uart_tx_buffer[uart_id]) = self->write_buffer.buf;
+
+        // set the irq handler
+        irq_set_exclusive_handler(self->irq_nr, self->irq_handler);
```

Comment:
This could be:
```
if (self->uart_id == 0) {
    irq_set_exclusive_handler(UART0_IRQ, uart0_irq_handler);
} else {
    irq_set_exclusive_handler(UART1_IRQ, uart1_irq_handler);
}
```
which would save some RAM not having to have these variables in the instance.  (Alternatively have a separate static table with irq number and handler.)

Reply:
That looks good and saves RAM.